searchings/linear_search.nim

```diff
@@ -0,0 +1,68 @@
+## Linear Search in Pure Nim
+## =============
+#[
+Linear search is the simplest but least efficient algorithm
+to search for an element in a data set.
+It examines each element until it finds a match,
+starting at the beginning of the data set, until the end.
+The search is finished and terminated once the target element is located.
+# https://www.simplilearn.com/tutorials/data-structure-tutorial/linear-search-algorithm
+]#
+## Worst case time complexity: O(N)
+## Average case time complexity: O(N)
+## Best case time complexity: O(1)
```

Comment:
Define above what you understand by worst, average, best, just to be sure.

Reply:
Sorry, it was an inaccuracy. No matter whether the inputs of an algorithm result in different runtime/efficiency/actual time, the time complexity does not change. It is still O(n). However What I meant by `worst case time complexity` is just the worst case the time of the algorithm can be, resulted by the input. Those two lines can actually be removed, as they are not actually helpful for users. Developers can figure out what the worst case and best case are, rightaway when they see the time complexity. I will change!